util-core/src/main/scala/com/twitter/util/Promise.scala

```diff
@@ -841,21 +841,41 @@ class Promise[A] extends Future[A] with Promise.Responder[A] with Updatable[Try[
   }
 
   @tailrec
-  protected final def continue(k: K[A]): Unit = state match {
-    case waitq: WaitQueue[A] =>
-      if (!cas(waitq, WaitQueue(k, waitq)))
-        continue(k)
-    case s: Interruptible[A] =>
-      if (!cas(s, new Interruptible(WaitQueue(k, s.waitq), s.handler, s.saved)))
-        continue(k)
-    case s: Transforming[A] =>
-      if (!cas(s, new Transforming(WaitQueue(k, s.waitq), s.other)))
-        continue(k)
-    case s: Interrupted[A] =>
-      if (!cas(s, new Interrupted(WaitQueue(k, s.waitq), s.signal)))
-        continue(k)
-    case v: Try[A] /* Done */ => k.runInScheduler(v)
-    case p: Promise[A] /* Linked */ => p.continue(k)
+  protected final def continue(k: K[A]): Unit = {
+    CoverageChecker.initialize("continue", 10)
+    state match {
+      case waitq: WaitQueue[A] =>
+        CoverageChecker.reached("continue", 0)
+        if (!cas(waitq, WaitQueue(k, waitq))) {
+          CoverageChecker.reached("continue", 1)
+          continue(k)
+        }
```

Comment:
There are a lot of implied `else` statements here that I think we should add coverage checks to. This would make it into 
```
if (!cas(waitq, WaitQueue(k, waitq))) {
   CoverageChecker.reached("continue", 1)
   continue(k)
} else {
    CoverageChecker.reached(...)
}
```

Reply:
This applies to all of the if statements :)

Reply:
Okay! I added coverage checkers in else conditions :)